rclcpp/test/rclcpp/test_create_timer.cpp

```diff
@@ -193,3 +193,44 @@ TEST(TestCreateTimer, timer_function_pointer)
 
   rclcpp::shutdown();
 }
+
+TEST(TestCreateTimer, timer_triggered_twice)
+{
+  rclcpp::init(0, nullptr);
+  auto node = std::make_shared<rclcpp::Node>("test_timer_triggered_twice");
+
+  std::atomic<int> callback_counter{0};
+
+  rclcpp::TimerBase::SharedPtr timer;
+  timer = rclcpp::create_timer(
+    node,
+    node->get_clock(),
+    rclcpp::Duration(0ms),
+    [&callback_counter]() {
+      callback_counter += 1;
+    }, nullptr, 2);
+
+  rclcpp::spin_some(node);
+  ASSERT_EQ(1, callback_counter);
+
+  rclcpp::spin_some(node);
+  ASSERT_EQ(2, callback_counter);
+
+  rclcpp::spin_some(node);
+  ASSERT_NE(3, callback_counter);
```

Comment:
```suggestion
  ASSERT_EQ(2, callback_counter); // not incremented
```
I think the assertion here should also be that `callback_counter` is equal to 2. The current assertion would succeed if the counter were equal to 4, but that is not the expected bahaviour.

Reply:
right, that is precise expectation here to check.

Reply:
I agree, this suggestion actually makes a lot more sense. Thanks and done.